### PR TITLE
Implementation UnitOfWork + Repository patterns

### PR DESCRIPTION
**Pull request by ShortQuality**

### CODE REVIEW

**Review by Tiktack — CHANGES_REQUESTED** [review]
Delete build files..........

**Review by isalzh — CHANGES_REQUESTED** [review]

File: src/Lab.Quiz.DAL/Interfaces/Repositories/IRepository.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Lab.Quiz.DAL.Interfaces
+{
+    public interface IRepository<TEntity> where TEntity : class
+    {
+        Task<TEntity> GetByIdAsync(Guid id);
+        IQueryable<TEntity> GetAll();
+        void AddAsync(TEntity entity);
```
Comment (Tiktack) [inline src/Lab.Quiz.DAL/Interfaces/Repositories/IRepository.cs:12]: Delete Async in name

**Review by Tiktack — CHANGES_REQUESTED** [review]

File: src/Lab.Quiz.DAL/Repositories/Repository.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using Lab.Quiz.DAL.Entities;
+using System.Collections.Generic;
+
+namespace Lab.Quiz.DAL.Repositories
+{
+    public class Repository<TEntity> : IRepository<TEntity> where TEntity : class 
+    {
+        private DbContext _dbContext;
+        DbSet<TEntity> Entities { get; set; }
+        public Repository(DbContext dbContext)
```
Comment (isalzh) [inline src/Lab.Quiz.DAL/Repositories/Repository.cs:15]: Again, there is no need to have a DbContext dependency in this class. As far as it is going to use only DbSet's things, just inject DbSet<TEntity> directly.

**Review by isalzh — COMMENTED** [review]
The UoW looks much better now

File: src/Lab.Quiz.DAL/Repositories/Repository.cs (hunk 2 above)
Comment (ShortQuality) [reply src/Lab.Quiz.DAL/Repositories/Repository.cs]: Like this?
image

File: src/Lab.Quiz.DAL/Repositories/Repository.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        DbSet<TEntity> Entities { get; set; }
+        public Repository(DbSet<TEntity> entities)
+        {
+            Entities = entities;
+        }
+
+        public async Task<TEntity> GetByIdAsync(Guid id)
+        {
+            return await Entities.FindAsync(id);
```
Comment (Tiktack) [inline src/Lab.Quiz.DAL/Repositories/Repository.cs:21]: In such cases like this method we actually can omit await and async. Anyway the code that will use this method will await it. But it's not very big change, so you can leave it as is.

File: src/Lab.Quiz.DAL/Repositories/Repository.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        public IQueryable<TEntity> GetAll()
+        {
+            return Entities;
+        }
+
+        public void Add(TEntity entity)
+        {
+            Entities.AddAsync(entity);
```
Comment (Tiktack) [inline src/Lab.Quiz.DAL/Repositories/Repository.cs:31]: MSDN documentation:
This method is async only to allow special value generators, such as the one used by 'Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo', to access the database asynchronously. For all other cases the non async method should be used.

File: src/Lab.Quiz.DAL/UnitOfWork.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Lab.Quiz.DAL.Entities;
+using Lab.Quiz.DAL.Interfaces;
+using Lab.Quiz.DAL.Repositories;
+
+namespace Lab.Quiz.DAL
+{
+    public sealed class UnitOfWork : IUnitOfWork ,IDisposable
```
Comment (Tiktack) [inline src/Lab.Quiz.DAL/UnitOfWork.cs:11]: space should be after comma, not before

**Review by Tiktack — APPROVED** [review]

File: src/Lab.Quiz.DAL/Repositories/Repository.cs (hunk 4 above)
Comment (ShortQuality) [reply src/Lab.Quiz.DAL/Repositories/Repository.cs:31]: Сhange it to non-async?

File: src/Lab.Quiz.DAL/Repositories/Repository.cs (hunk 3 above)
Comment (ShortQuality) [reply src/Lab.Quiz.DAL/Repositories/Repository.cs:21]: Ok, I got it

**Review by isalzh — APPROVED** [review]

**Pull request merged**